wrappers/hnygingonic/hnygingonic_example_router_test.go

```diff
@@ -0,0 +1,79 @@
+package hnygingonic
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/honeycombio/beeline-go"
+)
+
+func ExampleMiddleware() {
+	// Setup a new gin Router, not using the Default here so that we can put the
+	// Beeline middleware in before the middleware provided by Gin
+	router := gin.New()
+	router.Use(
+		// Add the beeline middleware to the chain
+		Middleware(nil),
+		// Doing something like the following would have the Middleware grab specifc
+		// GET query params that you deal with in your gin application.
+		//Middleware(map[string]struct{}{
+		//"parts":  {},
+		//"limit":  {},
+		//"offset": {},
+		//})
+		// The Logger and Recovery middleware which are setup in the Default gin router
+		gin.Logger(),
+		gin.Recovery(),
+		// Our example middleware that does extra work
+		exampleMiddleware(),
+	)
+
+	// Setup the routes we want to use
+	router.GET("/", home)
+	router.GET("/alive", alive)
+	router.GET("/ready", ready)
+
+	// Start the server
+	log.Fatal(router.Run("127.0.0.1:8080"))
+}
+
+func home(c *gin.Context) {
+	hnyctx, span := StartSpan(c, "main.home")
+	defer span.Send()
+	span.AddField("Welcome", "Home")
+	childFunction(hnyctx)
+	c.Data(http.StatusOK, "text/plain", []byte(`Welcome Home`))
+}
+
+func alive(c *gin.Context) {
+	_, span := StartSpan(c, "main.alive")
+	defer span.Send()
```

Comment:
I think I would leave out these two lines here. The middleware has already created a root span - these lines will add a child span. Which is fine, when it's useful, and it's nice to have the example that you can still create your own spans (and how) in the "complicated" handlers like `hme` or `ready`, but it's also fine to leave them out. For the health check leaving it out would be a good way of saying "yeah, we really don't need it here."

Reply:
Thanks :+1: I have updated the example.